### PR TITLE
Build and push Wasm OCI images.

### LINKED ISSUE #69: Package built binary as OCI images, push to GitHub Container Registry
Then we can leverage these built OCI images in the Istio tests (in addition to the fake ones as I did in the unit tests https://github.com/istio/istio/blob/master/pkg/wasm/imagefetcher_test.go#L80-L103). I've already done this in Proxy-Wasm Go SKD repository (https://github.com/orgs/tetratelabs/packages/container/package/proxy-wasm-go-sdk-examples).

Note that GHCR is free for OSS projects as long as the registry is public.

I can take care of this if it's ok. wdyt? @bianpengyuan

### PR DESCRIPTION
**Pull request by mathetake**
resolves #69

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -68,6 +72,21 @@ jobs:
         cd example
         bazel build //:example.wasm
 
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GHCR_TOKEN }}
```
Comment (mathetake) [inline .github/workflows/test.yaml:80]: looks like the `secrets.GHCR_TOKEN` is not configured even in the branches in this repository. Can you check if it is enabled? @bianpengyuan 

img

Comment (mathetake) [reply .github/workflows/test.yaml:80]: Oh I thought it will be automatically configured, but maybe someone has to set manually :D

Comment (bianpengyuan) [reply .github/workflows/test.yaml:80]: I added the secret and looks like login succeeded.

**Comment by mathetake** [comment]
```
2021/11/18 01:37:53 Error pushing image to ghcr.io/istio-ecosystem/wasm-extensions/scaffold:b5277dc782948d1d573ea076dc4acc6523aaa7a7: unable to push image to ghcr.io/istio-ecosystem/wasm-extensions/scaffold:b5277dc782948d1d573ea076dc4acc6523aaa7a7: POST https://ghcr.io/v2/istio-ecosystem/wasm-extensions/scaffold/blobs/uploads/: DENIED: permission_denied: create_package
```

@bianpengyuan can you make sure that the credential has the ^^ create_package permission ?

**Comment by bianpengyuan** [comment]
@mathetake yeah I did. Tested locally and I don't have permission to write package either. Looks like the package write permission is org scoped? @linsun could you grant me and/or @mathetake permission to create/delete packages at istio-ecosystem org?

**Comment by linsun** [comment]
@bianpengyuan you are already an admin in the wasm-extensions repo.  I also made @mathetake the admin as well.

**Comment by mathetake** [comment]
```
2021/11/23 23:48:25 Successfully pushed OCI image to ghcr.io/istio-ecosystem/wasm-extensions/basic_auth:b5277dc782948d1d573ea076dc4acc6523aaa7a7
```
now succeeded to push images, but I don't have permission to make the registries visible in (seems I need to have org-wide pkg related permissions) https://github.com/orgs/istio-ecosystem/packages

**Comment by bianpengyuan** [comment]
Yeah after I join the org, my token is able to push packages now. Looks like to make these image public, what we need to do is at org level `Settings > Packages`, tick the `Public`. After that I should be able to make these images public visible. Then we can turn it back to Private only. I will coordinate with @linsun offline on this.

**Comment by mathetake** [comment]
I think you can configure the visibility at registries level on "Package settings" in a package view. For example, the bottom right in this screen shot: 
Screen Shot 2021-11-24 at 10 11 49

**Comment by bianpengyuan** [comment]
@mathetak Yeah but currently it cannot be done and I think we need to change the org level package setting to allow public package?

Screen Shot 2021-11-23 at 5 16 54 PM
e

**Comment by mathetake** [comment]
oh I see! Thanks!

**Review by bianpengyuan — APPROVED** [review]

**Comment by bianpengyuan** [comment]
All packages are public now.

**Pull request merged**